postr/facebook_api.py

```diff
@@ -0,0 +1,151 @@
+# Facebook API
+# from typing import List
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import webbrowser
+import json
+from typing import Type
+import config
+import facebook
+from api_interface import ApiInterface
+
+code = ''
+
+
+class Handler(BaseHTTPRequestHandler):
+
+    def do_GET(self) -> None:
+        request_path = self.path
+        global code  # pylint: disable=global-statement
+        self.send_response(200)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+        code = request_path
+        output = '<html><head><title>Postr Facebook Auth</title></head>'
+        output += '<body><h1>Thanks For Authenticationg with Postr!</h1>'
+        output += '<p>You can close this tab.</p>'
+        output += '</body></html>'
+        self.wfile.write(output.encode())
+
+
+class Server(HTTPServer):
+    pass
+
+
+class FacebookApi(ApiInterface):
+
+    def __init__(self) -> None:
+
+        if config.get_api_key('FACEBOOK', 'has_token') == 'true':
+
+            authToken = config.get_api_key('FACEBOOK', 'auth_token')
+            self.graph = facebook.GraphAPI(
+                access_token=authToken,
+                version='2.12',
+            )
+        else:  # may need to just auth user with no has_token check
+            # need to auth  user
+            FacebookApi.authenticate()
+
+            authToken = config.get_api_key('FACEBOOK', 'authToken')
+            self.graph = facebook.GraphAPI(
+                access_token=authToken,
+                version='2.12',
+            )
+
+    @staticmethod
+    def wait_for_request(
+        server_class: Type[HTTPServer],  # pylint: disable=bad-continuation
```

Comment:
don't think we need the pylint marks now that it's disabled in the global config